webapp/app/js/perf.js

```diff
@@ -231,52 +234,71 @@ perf.factory('PhCompare', [ '$q', '$http', 'thServiceDomain', 'PhSeries',
       return cr.className;
     },
 
-    getCounterMap: function(testName, originalData, newData) {
+
+
+    // Aggregates two sets of values into a "comparison object" which is later used
+    // to display a single line of comparison at compareperf.js .
+    getCounterMap: function getDisplayLineData(testName, originalData, newData) {
+
+      // Some statistics for a single set of values
+      function analyzeSet(values) {
+        var stddev = math.stddev(values),
+            geomean = math.geomean(values);
+
+        return {
+          runs: values.length,
+          geomean: geomean,
+          stddev: stddev,
+          stddevPct: math.percentOf(stddev, geomean)
+        };
+      }
+
       var cmap = {originalGeoMean: 0, originalRuns: 0, originalStddev: 0,
-                  newGeoMean: 0, newRuns: 0, newStddev: 0, delta: 0,
-                  deltaPercentage: 0, barGraphMargin: 0, isEmpty: false,
-                  isRegression: false, isImprovement: false, isMinor: true};
+                  newGeoMean:      0, newRuns:      0, newStddev:      0,
+                  delta: 0, deltaPercentage: 0, barGraphMargin: 0,
+                  isEmpty: false, isRegression: false, isImprovement: false, isMinor: true};
 
+      // Data on each set
+
+      // cmap.*Runs is the number of runs plus the runs values for display.
       if (originalData) {
-         cmap.originalGeoMean = originalData.geomean;
-         cmap.originalRuns = originalData.runs;
-         cmap.originalStddev = originalData.stddev;
-         cmap.originalStddevPct = ((originalData.stddev / originalData.geomean) * 100);
-         cmap.originalMin = originalData.minVal;
-         cmap.originalMax = originalData.maxVal;
+        var orig = analyzeSet(originalData.values);
+        cmap.originalGeoMean = orig.geomean;
+        cmap.originalRuns = orig.runs + "  <  " + originalData.values.join("   ") + "  >";;
```

Comment:
>  The .runs value is not used for anything else as far as I can tell.

That's actually wrong - it's used few lines below to determine if a set is empty. But this is sort of a bug at the original code too, since if a set is empty we shouldn't have tried to calculate it's statistics to begin with. I'll fix both.